### PR TITLE
Add code examples for all the new explainability plotting functions in Python and R (PUBDEV-7824, PUBDEV-7821)

### PR DESCRIPTION
**Pull request by tomasfryda**
I added examples to the python and R docstrings for the public functions/methods. 

Related JIRAs:
https://0xdata.atlassian.net/browse/PUBDEV-7824
https://0xdata.atlassian.net/browse/PUBDEV-7821

### CODE REVIEW

**Comment by ledell** [comment]
@tomasfryda I noticed in [`def explain`](https://github.com/h2oai/h2o-3/blob/32b0a826594081d0b9ddf6ddcb89cbacf5c675bd/h2o-py/h2o/explanation/_explain.py#L1942) and [`def explain_row`](https://github.com/h2oai/h2o-3/blob/32b0a826594081d0b9ddf6ddcb89cbacf5c675bd/h2o-py/h2o/explanation/_explain.py#L2214), that it says `models` is `H2OAutoML object or H2OModel`.  Should we add "list of models" here?  This is officially supported, right?


```
    :param models: H2OAutoML object or H2OModel
```

**Comment by ledell** [comment]
It's helpful in examples to use generic arg names as possible to make it easy for people to swap out our dataset with theirs.  Here's some changes I would make:

```
titanic = h2o.import_file("https://h2o-public-test-data.s3.amazonaws.com/smalldata/titanic/titanic_expanded.csv")
```
becomes:
```
f = "https://h2o-public-test-data.s3.amazonaws.com/smalldata/titanic/titanic_expanded.csv"
df = h2o.import_file(f)
```
If you can avoid hardcoding the predictor columns, I think that would be a slight improvement too.  (They are hardcoded in Py docstrings but not R).

Also, since we are only trying to demo `h2o.explain()` instead of AutoML, I would lower the number of models in the AutoML run to something minimal (where SE still wins).  Maybe for consistency, we could just use the same exact example from the User Guide.  The Wine example?

**Comment by tomasfryda** [comment]
> @tomasfryda I noticed in [`def explain`](https://github.com/h2oai/h2o-3/blob/32b0a826594081d0b9ddf6ddcb89cbacf5c675bd/h2o-py/h2o/explanation/_explain.py#L1942) and [`def explain_row`](https://github.com/h2oai/h2o-3/blob/32b0a826594081d0b9ddf6ddcb89cbacf5c675bd/h2o-py/h2o/explanation/_explain.py#L2214), that it says `models` is `H2OAutoML object or H2OModel`. Should we add "list of models" here? This is officially supported, right?
> 
> ```
>     :param models: H2OAutoML object or H2OModel
> ```

Yes, they are supported, I will fix that tomorrow.

**Comment by tomasfryda** [comment]
> It's helpful in examples to use generic arg names as possible to make it easy for people to swap out our dataset with theirs. Here's some changes I would make:
> 
> ```
> titanic = h2o.import_file("https://h2o-public-test-data.s3.amazonaws.com/smalldata/titanic/titanic_expanded.csv")
> ```
> 
> becomes:
> 
> ```
> f = "https://h2o-public-test-data.s3.amazonaws.com/smalldata/titanic/titanic_expanded.csv"
> df = h2o.import_file(f)
> ```
> 
> If you can avoid hardcoding the predictor columns, I think that would be a slight improvement too. (They are hardcoded in Py docstrings but not R).
> 
> Also, since we are only trying to demo `h2o.explain()` instead of AutoML, I would lower the number of models in the AutoML run to something minimal (where SE still wins). Maybe for consistency, we could just use the same exact example from the User Guide. The Wine example?

I can lower the amount of the models in AutoML, but it is good to note that `varimp_heatmap` doesn't work for stacked ensembles so I would go with at least 7 models, so it looks nice (5 models in the `varimp_heatmap`).

**Comment by ledell** [comment]
R CMD check is currently failing with a valid error:

```
[2020-10-05T21:37:08.996Z] > pdp <- h2o.pd_multi_plot(aml, test, column = "age")
[2020-10-05T21:37:08.996Z] Error in h2o.pd_multi_plot(aml, test, column = "age") : 
[2020-10-05T21:37:08.996Z]   Column was not found in the provided data set!
[2020-10-05T21:37:08.996Z] Execution halted
```

The dataset was updated to Wine data, but still using Titanic colname.  I'll fix.

**Review by ledell — APPROVED** [review]
Tests passing now, thanks!

File: h2o-py/h2o/explanation/_explain.py [hunk 1]
```diff
@@ -876,7 +923,30 @@ def pd_plot(
...
+    >>>
+    >>> # Split the dataset into a train and test set:
+    >>> train, test = df.split_frame([.75])
+    >>>
+    >>> # Train a GBM
+    >>> gbm = H2OGradientBoostingEstimator()
+    >>> gbm.train(y=response, training_frame=train)
+    >>>
+    >>> # Create partial dependence plot
+    >>> gbm.pd_plot(test, column="age")
```
Comment (ledell) [inline h2o-py/h2o/explanation/_explain.py]:
```suggestion
    >>> gbm.pd_plot(test, column="alcohol")
```

File: h2o-py/h2o/explanation/_explain.py [hunk 2]
```diff
@@ -974,7 +1044,30 @@ def pd_multi_plot(
...
+    >>>
+    >>> # Split the dataset into a train and test set:
+    >>> train, test = df.split_frame([.75])
+    >>>
+    >>> # Train an H2OAutoML
+    >>> aml = H2OAutoML(max_models=7)
+    >>> aml.train(y=response, training_frame=train)
+    >>>
+    >>> # Create a partial dependence plot
+    >>> aml.pd_multi_plot(test, column="age")
```
Comment (ledell) [inline h2o-py/h2o/explanation/_explain.py]:
```suggestion
    >>> aml.pd_multi_plot(test, column="alcohol")
```

**Review by ledell — APPROVED** [review]
I just made a few changes to the pydocs (one bug, one two edits to match R).

**Comment by ledell** [comment]
Tests already passed.  Thanks @tomasfryda!

**Pull request merged**